Apps/UnitTests/Scripts/tests.js

```diff
@@ -327,6 +327,110 @@ describe("PostProcesses", function() {
+        setTimeout(() => {
+            try {
+                expect(new Date().getTime() - startTime).to.be.at.least(10);
+            }
+            catch (e) {
+                done(e);
+                return;
+            }
+            done();
+        }, 10);
+    });
+    it("should call the given function after the given delay when the delay is a string representing a valid number", function (done) {
+        const startTime = new Date().getTime();
+        setTimeout(() => {
+            try {
+                expect(new Date().getTime() - startTime).to.be.at.least(10);
+            }
+            catch (e) {
+                done(e);
+                return;
+            }
+            done();
+        }, "10");
+    });
+    it("should call the given function after zero milliseconds when the delay is a string representing an invalid number", function (done) {
+        setTimeout(() => {
+            done();
+        }, "a");
+    });
+    it("should call the given function after other tasks execute when the given delay is zero", function (done) {
+        let trailingCodeExecuted = false;
+        setTimeout(() => {
+            try {
+                expect(trailingCodeExecuted).to.be.true;
+            }
+            catch (e) {
+                done(e);
+                return;
+            }
+            done();
+         }, 0);
+        trailingCodeExecuted = true
+    });
+    it("should call the given function after other tasks execute when the given delay is undefined", function (done) {
+        let trailingCodeExecuted = false;
+        setTimeout(() => {
+            try {
+                expect(trailingCodeExecuted).to.be.true;
+            }
+            catch (e) {
+                done(e);
+                return;
+            }
+            done();
+         }, undefined);
+        trailingCodeExecuted = true
+    });
+    it("should call the given functions in the correct order", function (done) {
+        let called = []
+        for (let i = 9; i >= 0; i--) {
+            setTimeout(() => {
+                called.push(i * 2);
+                if (called.length === 10) {
+                    try {
+                        expect(called).to.deep.equal([0, 2, 4, 6, 8, 10, 12, 14, 16, 18]);
+                    }
+                    catch (e) {
+                        done(e);
+                        return;
+                    }
+                    done();
+                }
+            }, i * 2);
+        }
+    });
+})
+
+describe("clearTimeout", function () {
+    this.timeout(0);
+    it("should stop the timeout matching the given timeout id", function (done) {
+        const id = setTimeout(() => {
+            done(new Error("Timeout was not cleared"));
+        }, 0);
+        clearTimeout(id);
+        setTimeout(done, 100);
+    });
+    it("should do nothing if the given timeout id is undefined", function (done) {
+        setTimeout(() => { done(); }, 0);
```

Comment:
nit: is this necessary? can you just call `done()` after `clearTimeout`

Reply:
`done()` is called in the `setTimeout` callback to make sure `clearTimeout` has not cleared it.

Reply:
that seems arbitrary, but okay.

Reply:
Ya, it's not very useful but it's not taking up much time beyond making sure `clearTimeout(undefined)` doesn't crash.